azurerm/resource_arm_virtual_machine.go

```diff
@@ -721,91 +721,90 @@ func resourceArmVirtualMachineRead(d *schema.ResourceData, meta interface{}) err
-
-	if resp.VirtualMachineProperties.StorageProfile.ImageReference != nil {
-		if err := d.Set("storage_image_reference", schema.NewSet(resourceArmVirtualMachineStorageImageReferenceHash, flattenAzureRmVirtualMachineImageReference(resp.VirtualMachineProperties.StorageProfile.ImageReference))); err != nil {
-			return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage Image Reference error: %#v", err)
+	if props := resp.VirtualMachineProperties; props != nil {
+		if availabilitySet := props.AvailabilitySet; availabilitySet != nil {
+			// TODO: why is this being lower-cased?
+			d.Set("availability_set_id", strings.ToLower(*availabilitySet.ID))
 		}
-	}
 
-	if osDisk := resp.VirtualMachineProperties.StorageProfile.OsDisk; osDisk != nil {
-		diskInfo, err := resourceArmVirtualMachineGetManagedDiskInfo(osDisk.ManagedDisk, meta)
-		if err != nil {
-			return fmt.Errorf("[DEBUG] Error getting managed OS disk detailed information: %#v", err)
-		}
-		if err := d.Set("storage_os_disk", flattenAzureRmVirtualMachineOsDisk(osDisk, diskInfo)); err != nil {
-			return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage OS Disk error: %#v", err)
+		if profile := props.HardwareProfile; profile != nil {
+			d.Set("vm_size", profile.VMSize)
 		}
-	}
 
-	if dataDisks := resp.VirtualMachineProperties.StorageProfile.DataDisks; dataDisks != nil {
-		disksInfo := make([]*compute.Disk, len(*dataDisks))
-		for i, dataDisk := range *dataDisks {
-			diskInfo, err := resourceArmVirtualMachineGetManagedDiskInfo(dataDisk.ManagedDisk, meta)
-			if err != nil {
-				return fmt.Errorf("[DEBUG] Error getting managed data disk detailed information: %#v", err)
+		if profile := props.StorageProfile; profile != nil {
+			if err := d.Set("storage_image_reference", schema.NewSet(resourceArmVirtualMachineStorageImageReferenceHash, flattenAzureRmVirtualMachineImageReference(profile.ImageReference))); err != nil {
+				return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage Image Reference error: %#v", err)
 			}
-			disksInfo[i] = diskInfo
-		}
-		if err := d.Set("storage_data_disk", flattenAzureRmVirtualMachineDataDisk(dataDisks, disksInfo)); err != nil {
-			return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage Data Disks error: %#v", err)
-		}
-	}
 
-	if resp.VirtualMachineProperties.OsProfile != nil {
-		if err := d.Set("os_profile", schema.NewSet(resourceArmVirtualMachineStorageOsProfileHash, flattenAzureRmVirtualMachineOsProfile(resp.VirtualMachineProperties.OsProfile))); err != nil {
-			return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage OS Profile: %#v", err)
-		}
+			if osDisk := profile.OsDisk; osDisk != nil {
+				diskInfo, err := resourceArmVirtualMachineGetManagedDiskInfo(osDisk.ManagedDisk, meta)
+				if err != nil {
+					return fmt.Errorf("Error flattening `storage_os_disk`: %#v", err)
+				}
+				if err := d.Set("storage_os_disk", flattenAzureRmVirtualMachineOsDisk(osDisk, diskInfo)); err != nil {
```

Comment:
I don't think that the `err` needs to be re-initialized here since it was already defined above in line 748, what do you think?

```
if err = d.Set("storage_os_disk", flattenAzureRmVirtualMachineOsDisk(osDisk, diskInfo)); err != nil {
```

Reply:
whilst technically true at this point we're not reusing the variable `err` below within the inner scope - so I don't think it's a big deal either way?

Reply:
going to merge this either way, we can fix this up later if needed